src/stack.ts

```diff
@@ -1436,11 +1436,12 @@ export class Stack {
 
     schemas.forEach((schema) => {
       // Entry references
-      entryReferences = mergeWith(entryReferences, schema[this.types.references])
-      // tslint:disable-next-line: forin
-      for (const path in schema[this.types.assets]) {
-        paths.push(path)
-      }
+      entryReferences = _.mergeWith(entryReferences, schema[this.types.references], (existingReferences, newReferences) => {
+        if (_.isArray(existingReferences)) { 
+          return Array.from(new Set(existingReferences.concat(newReferences))); 
+        }
+        return existingReferences;
+      });
```

Comment:
@harshithad0703 , Importing lodash in the manner of this import _ from 'lodash' is not necessary. You can add methods directly because we've already imported it line no 10.

Reply:
ok